xenvm: dom0: update Xen public headers for running Zephyr as Dom0

xakep-amatop opened the pull request:
Xen public headers are added to Zephyr kernel partially, so we need to update them for new features implementation.

Further implementation of Xen domains configuration and memory management requires dedicated Xen public headers inside Zephyr kernel. It will be used for add Zephyr OS Xen Domain-0 funtionalities.

Update existent and add new required public headers from Xen 4.19.0 release.

xakep-amatop committed:
xenvm: dom0: update Xen public headers for running Zephyr as Dom0

Xen public headers are added to Zephyr kernel partially, so we need
to update them for new features implementation.

Further implementation of Xen domains configuration and memory
management requires dedicated Xen public headers inside Zephyr kernel.
It will be used for add Zephyr OS Xen Domain-0 funtionalities.

Update existent and add new required public headers from Xen 4.18.2
release.

Signed-off-by: Mykola Kvach <mykola_kvach@epam.com>

xakep-amatop commented on include/zephyr/xen/public/arch-arm.h:
```diff
@@ -312,14 +351,26 @@ DEFINE_XEN_GUEST_HANDLE(vcpu_guest_context_t);
 	uint16_t tee_type;
 	/* IN */
 	uint32_t nr_spis;
+#if __XEN_INTERFACE_VERSION__ >= 0x00041300
```
Note: actually, there aren't these changes in mainline
it is EPAM related: https://github.com/xen-troops/xen/commit/d4192ca563d80ca775eadbce278eccc3ebc1bc8e#diff-f7cea1bef753331288b1f2c3e15a1d9a6f6e4e48ef59cfabff177d185c8503a8R343

xakep-amatop commented:
note: all errors related to MIT licenses or adding new typedefs

firscity commented:
> new required public headers from Xen 4.19.0 release.

Is the 4.19.0 release is actually available? AFAIK, it is still in progress

xakep-amatop commented:
> > new required public headers from Xen 4.19.0 release.
> 
> Is the 4.19.0 release is actually available? AFAIK, it is still in progress

I've synced it with the latest branch used in prod devel

GrygiriiS commented:
Why not just copy files from Xen -> Zephyr? at least some of them can be just copied as is? may be with licence changes.
Or at least could you please insert new definitions at the same places as they are in Xen, so diff viewers can be used easy to see actual diffs Xen vs Zephyr?

For example: event_channel.h you've added all changes at the end which makes not easy (or very very hard) to track actual state and differences Xen vs Zephyr by using any diff viewers.

xakep-amatop commented:
> Why not just copy files from Xen -> Zephyr? at least some of them can be just copied as is? may be with licence changes. Or at least could you please insert new definitions at the same places as they are in Xen, so diff viewers can be used easy to see actual diffs Xen vs Zephyr?
> 
> For example: event_channel.h you've added all changes at the end which makes not easy (or very very hard) to track actual state and differences Xen vs Zephyr by using any diff viewers.

I don't understand your point if you compare xen public dir files with files in zephyr public dir they have minimal differences, lets compare file that you mention:
```
$ diff -w xen/include/public/event_channel.h ../zephyr/zephyr/include/zephyr/xen/public/event_channel.h
1a2
> 
6a8,25
>  * Permission is hereby granted, free of charge, to any person obtaining a copy
>  * of this software and associated documentation files (the "Software"), to
>  * deal in the Software without restriction, including without limitation the
>  * rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
>  * sell copies of the Software, and to permit persons to whom the Software is
>  * furnished to do so, subject to the following conditions:
>  *
>  * The above copyright notice and this permission notice shall be included in
>  * all copies or substantial portions of the Software.
>  *
>  * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
>  * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
>  * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
>  * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
>  * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
>  * FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
>  * DEALINGS IN THE SOFTWARE.
>  *
37,45d55
< /*
<  * ` enum neg_errnoval
<  * ` HYPERVISOR_event_channel_op(enum event_channel_op cmd, void *args)
<  * `
<  * @cmd  == EVTCHNOP_* (event-channel operation).
<  * @args == struct evtchn_* Operation-specific extra arguments (NULL if none).
<  */
< 
< /* ` enum event_channel_op { // EVTCHNOP_* => struct evtchn_* */
63d72
< /* ` } */
328d336
< 
330d337
< 
362,371d368
< 
< /*
<  * Local variables:
<  * mode: C
<  * c-file-style: "BSD"
<  * c-basic-offset: 4
<  * tab-width: 4
<  * indent-tabs-mode: nil
<  * End:
<  */
```

as you can see changes only related to license headers and some comments
all headers in Zephyr as much possible matches Xen headers with only some style chages and with changes of some defines

xakep-amatop committed:
xenvm: dom0: update Xen public headers for running Zephyr as Dom0

Xen public headers are added to Zephyr kernel partially, so we need
to update them for new features implementation.

Update existent and add new required public headers from Xen main branch
and some of changes from xen-troops xen repo.

Signed-off-by: Mykola Kvach <mykola_kvach@epam.com>

xakep-amatop commented:
Split up the changes into two commits, update headers based on:
* the release 4.18.2 tag;
* xen mainline (also known as future 4.19.0) and xen-troops prod devel changes (https://github.com/xen-troops/xen/tree/xen-4.19-xt0.1).
